src/js/data-builder.js

```diff
@@ -105,13 +105,95 @@ const buildData = (json) => {
 
   return data;
 };
-const filterNodes = (json, nodes = []) => {
-  let filteredArray = json.filter((element) => {
+const filterObjects = (json, nodes = []) => {
+  let filteredJson = json.filter((element) => {
     return nodes.filter((node) => node === element.GRANTED_TO_NAME && element.GRANTED_TO_TYPE === 'ROLE').length > 0
       ? true
       : false;
   });
-  const x = buildData(filteredArray);
+  const x = buildData(filteredJson);
   return x.hierarchy;
 };
-export { buildData, filterNodes };
+
+const filterNodes = (formattedData, json, jstreeObject) => {
+  let filteredJson = [];
+  const jstreeData = jstreeObject.instance.get_json('#', { flat: true });
+  const selectedObj = jstreeObject.instance.get_node(jstreeObject.selected[0]);
+  let levels = JSON.parse(JSON.stringify(selectedObj.parents));
+  levels = levels.reverse();
+
+  let leafIndex = levels.findIndex((leaf) => leaf === selectedObj.id);
+  if (leafIndex === -1) {
+    levels.push(selectedObj.id);
+  }
+
+  let levelsData = [];
+  for (let levelIndex of levels) {
+    if (levelIndex !== '#') {
+      levelsData.push(jstreeData.filter((x) => x.id === levelIndex)[0]);
+    }
+  }
+
+  filteredJson = json.filter((element) => {
```

Comment:
some commenting would be really good here. Massive IF nests are ripe for errors without clean comments and good variable names